Morgan-Stanley/treadmill
# Sync with 2017.05.16-1

* More prints replaced with print function.
 * Add explanation to trace pending event.
 * Replace print with print_function.
 * watchdog: Disable as it causes spurous reboots
 * network service: Add support for external IP override
 * host_ip: Move host_ip to network service
 * Support up to 5 masters, this is needed for VI cells
 * Handle LDAP connection failure on app monitor.
 * add a name to the default partition
 * Fixed shared network flag support for application
 * Eventd to exit on zk disconnect.
 * Fixed and exposed the server list API, tests.
 * fix bug when loading allocations in scheduler
 * Changed app monitor algorithm.
 * logs cli: use websocket to get latest trace to figure out host and uniq id if not specified
 * add allocation delete command
 * tests: Style and correctness fixes
 * refactor: Rename `treadmill.tests` to `treadmill.checkout`
 * fix init / watchdog
 * add some documentation about allocations
 * Refactored admin install and bootstrap.
 * list partitions as part of cell details
 * Remove print.
 * Encode event data to ensure that it can be filename.
 * Refactored trace implementation.
 * Clean up the allocation cli and add capacity check before create
 * added location field to cli command list cell
